src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageCredentialsValidator.cs

```diff
@@ -56,9 +56,37 @@ internal class DefaultStorageCredentialsValidator : IStorageCredentialsValidator
             catch
             {
                 string message = String.Format(CultureInfo.CurrentCulture,
-                    "The account credentials for '{0}' are incorrect.", account.Credentials.AccountName);
+                    "Invalid storage account '{0}'. Please make sure your credentials are correct.",
+                    account.Credentials.AccountName);
                 throw new InvalidOperationException(message);
             }
+
+            if (isPrimaryAccount)
+            {
+                // Primary storage accounts require Queues
+                var queueClient = account.CreateQueueClient();
+                var queue = queueClient.GetQueueReference("name");
+                try
+                {
+                    await queue.ExistsAsync(cancellationToken);
+                    _primaryCredentials = account.Credentials;
+                }
+                catch (OperationCanceledException)
+                {
+                    throw;
+                }
+                catch (StorageException exception)
+                {
+                    if (exception.Message.Contains("The remote name could not be resolved"))
+                    {
+                        string message = String.Format(CultureInfo.CurrentCulture,
+                            "Invalid storage account '{0}'. Primary storage accounts must be general "
+                            + "purpose accounts and not restricted blob storage accounts.", account.Credentials.AccountName);
```

Comment:
Have you tested that an incorrect host/account in the connection string won't end up down this path? I expect that to be the case (as it will lead to the same failure) and if so, this message may be a bit misleading/confusing. We may need to update this verbiage, as without a reliable check against some account metadata, we can't assume that we failed because of a blob-only account.